pkg/operation/botanist/dns.go

```diff
@@ -86,6 +87,10 @@ func (b *Botanist) DestroyExternalDomainDNSRecord(ctx context.Context) error {
 }
 
 func (b *Botanist) deployDNSProvider(ctx context.Context, name, provider string, secretData map[string][]byte, includedDomains ...string) error {
+	if provider == v1beta1.DNSUnmanaged {
```

Comment:
Can we let `shoot_control_reconcile` rather skip the complete steps of deploying the internal dns entry? https://github.com/gardener/gardener/blob/b6e614486c3f72818592eb060592f2ec604ae3b8/pkg/controllermanager/controller/shoot/shoot_control_reconcile.go#L109.

like we do it for:
https://github.com/gardener/gardener/blob/b6e614486c3f72818592eb060592f2ec604ae3b8/pkg/controllermanager/controller/shoot/shoot_control_reconcile.go#L114